### PR TITLE
chore(ci): enable junit5

### LINKED ISSUE #5103: Investigate how to support JUnit 5 tests in Zeebe
**Description**

* we would like to write JUnit 5 tests
* we don't want to migrate all our existing tests to JUnit 5 right away - so the existing tests shall still run

### PR DESCRIPTION
**Pull request by pihme**
## Description

Adds JUnit5 support to Zeebe.

## Related issues

closes #5103

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [X] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the release announcement

### CODE REVIEW

**Comment by pihme** [comment]
@menski I was able to get rid of the exclusion. Please do the review when it suits you.

File: parent/pom.xml [hunk 1]
```diff
@@ -218,6 +229,13 @@
         <version>${version.mockito}</version>
       </dependency>
 
+      <dependency>
+        <groupId>org.mockito</groupId>
+        <artifactId>mockito-junit-jupiter</artifactId>
+        <version>${version.mockito-jupiter}</version>
+        <scope>test</scope>
```
Comment (menski) [inline parent/pom.xml:236]: I don't think we need the `test` scope here as this is dependency management.

**Review by menski — CHANGES_REQUESTED** [review]
The build is not passing because of opentest4j. If I find time I can have a look at it later today, or maybe you can check it out again. I think we are still missing the opentest4j dependency to pin the version in the dependency management. 

```
Dependency convergence error for org.opentest4j:opentest4j:1.1.1 paths to dependency are:
[2020-08-12T07:35:14.547Z] +-io.zeebe:zeebe-client-java:0.25.0-SNAPSHOT
[2020-08-12T07:35:14.547Z]   +-com.github.tomakehurst:wiremock-jre8:2.27.1
[2020-08-12T07:35:14.547Z]     +-net.javacrumbs.json-unit:json-unit-core:2.12.0
[2020-08-12T07:35:14.547Z]       +-org.opentest4j:opentest4j:1.1.1
[2020-08-12T07:35:14.547Z] and
[2020-08-12T07:35:14.547Z] +-io.zeebe:zeebe-client-java:0.25.0-SNAPSHOT
[2020-08-12T07:35:14.547Z]   +-org.junit.jupiter:junit-jupiter-engine:5.6.2
[2020-08-12T07:35:14.547Z]     +-org.junit.platform:junit-platform-engine:1.6.2
[2020-08-12T07:35:14.547Z]       +-org.opentest4j:opentest4j:1.1.1
[2020-08-12T07:35:14.547Z] and

[2020-08-12T07:35:14.547Z] +-io.zeebe:zeebe-client-java:0.25.0-SNAPSHOT
[2020-08-12T07:35:14.547Z]   +-org.junit.jupiter:junit-jupiter-engine:5.6.2
[2020-08-12T07:35:14.547Z]     +-org.junit.jupiter:junit-jupiter-api:5.6.2
[2020-08-12T07:35:14.547Z]       +-org.opentest4j:opentest4j:1.2.0
```

**Comment by pihme** [comment]
Why am I not seeing these dependency convergence warnings when I run `mvn clean install`?

**Review by menski — APPROVED** [review]
bors r+

**Pull request merged**